README.md

```diff
@@ -20,135 +14,123 @@ protocol ProtocolName {
+    tag = 'Object';
+  };
 }
-
-Promise.prototype[Functor.map] = function (f) {
-  return this.then(function(x) {
-    if (x instanceof Identity) {
-      x = x.unwrap();
-    }
-    let result = f.call(this, x);
-    if (result instanceof Promise) {
-      result = new Identity(result);
-    }
-    return result;
-  });
-};
-
-Protocol.implement(Promise, Functor);
 ```
 
-Finally, one of the biggest benefits of protocols is that they eliminate the
-fear of mutating built-in prototypes. One of the beautiful aspects of
-ECMAScript is its ability to extend its built-in prototypes. But with the
-limited string namespace, this is untenable in large codebases and impossible
-when integrating with third parties. Because protocols are based on symbols,
-this is no longer an anti-pattern.
+Protocols isolate implementations meant for different purposes by keeping them
+in a separate object with certain static guarantees.
+
+One of the biggest benefits of protocols is that they eliminate the
+fear of mutating built-in prototypes in potentially conflicting ways. One of the
+beautiful aspects of ECMAScript is its ability to extend its built-in
+prototypes. But with the limited string namespace, this is untenable in large
+codebases and impossible when integrating with third parties. Because protocols
+are based on symbols and support coherence, this is no longer an anti-pattern.
 
 ```js
 class Ordering {
-  static LT = new Ordering;
-  static EQ = new Ordering;
-  static GT = new Ordering;
+  constructor (str) { this.symbol = Symbol(str) }
 }
 
 protocol Ordered {
+  static LT = new Ordering('LT');
+  static EQ = new Ordering('EQ');
+  static GT = new Ordering('GT');
+  static fromNum (num) {
+      if (num < 0) return Ordered.LT;
+      if (num === 0) return Ordered.EQ;
+      if (num > 0) return Ordered.GT;
+    }
+  }
   compare;
 
   lessThan(other) {
-    return this[Ordered.compare](other) === Ordering.LT;
+    return this[Ordered].compare(other) === Ordered.LT;
+  }
+
+  implFor String {
+    compare(other) {
+      return Ordered.fromNum(String.localeCompare(other));
+    }
+  }
+
+  implFor Number {
+    compare(other) {...}
   }
 }
 
-String.prototype[Ordered.compare] = function() { /* elided */ };
-Protocol.implement(String, Ordered);
+(2)[Ordered].compare(1) // Ordered.GT
+'a'[Ordered].lessThan('aa') // true
 ```
 
-
-## Features
+## Other Features
 
 ### protocol inheritance
 
-Protocols may extend other protocols. This expresses a dependency
-relationship between the protocols.
+Protocols may extend other protocols. This expresses a dependency relationship
+between the protocols.
 
 ```js
-protocol A { a; }
-protocol B extends A { b; }
+protocol Iterator { next; }
+protocol PeekableIterator extends Iterator { prev; }
 
-class C implements B {
-  [A.a]() {}
-  [B.b]() {}
+class BackAndForth {
+  constructor (values) {
+    this.values = [...values]
+    this.idx = 0
+  }
+  // Iterator MUST be implemented if PeekableIterator is being defined.
```

Comment:
Can both methods be defined in the implements PeekabeIterator block, or must the inheritance tree be explicit on the consumer side?

Reply:
No, you -must- implement protocols separately. I think this is appropriate because it makes coherence restrictions more clear.